src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace NServiceBus
+{
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extentions for <see cref="EndpointConfiguration"/>.
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="endpointName">The endpoint name to be used.</param>
+        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
+        {
+            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
```

Comment:
We decided to not provide it on IConfigureThisEndpoint. So that option is
off the table. Regarding constants. Your argument is correct but with the
approvals we break the build when you change the consts cause they are part
of the api